orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -607,23 +607,25 @@ def run_step(model1, model2, model3, x1, x2):
 def test_ortmodule_inputs_with_dynamic_shape():
     D_in, H, D_out = 784, 500, 10
 
-    model = NeuralNetSinglePositionalArgument(D_in, H, D_out).to('cuda')
-    model = ORTModule(model)
+    pt_model = NeuralNetSinglePositionalArgument(D_in, H, D_out).to('cuda')
+    ort_model = ORTModule(copy.deepcopy(pt_model))
+
+    def run_step(model, x):
+        p = model(x)
+        loss = p.sum()
+        loss.backward()
+        return p
 
     for step in range(10):
         N = random.randint(1,100)
         x = torch.randn(N, D_in, device='cuda', requires_grad=True)
         assert x.grad is None
 
-        prediction = model(x)
-        s = prediction.sum()
-        s.backward()
-
-        assert x.grad is not None 
-        for param in model.parameters():
-            assert param.grad is not None
-            param.grad = None
+        pt_p = run_step(pt_model, x)
+        ort_p = run_step(ort_model, x)
 
+        assert torch.allclose(ort_p, pt_p, atol=1e-6)    # relaxing tolerance, 1e-7 or less would fail
```

Comment:
1e-6 is ok. 
see tol setting in assert_gradients_match_and_reset_gradient.

Reply:
cool, would this also help on other UTs with TODOs?